templates/terraform/aws_agent_with_opentelemetry_collector/main.tf

```diff
@@ -0,0 +1,92 @@
+# Monte Carlo's Agent with OpenTelemetry Collector - Terraform Configuration
+# Copyright 2023 Monte Carlo Data, Inc.
+
+terraform {
+  required_version = ">= 1.0"
+
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 5.0"
+    }
+  }
+}
+
+# Data sources
+data "aws_caller_identity" "current" {}
+data "aws_region" "current" {}
+
+# Local values for conditions (equivalent to CloudFormation Conditions)
+locals {
+  use_existing_telemetry_data_bucket = var.opentelemetry_collector_existing_bucket_arn != "N/A"
+  has_notification_channel           = var.opentelemetry_collector_external_notification_channel_arn != "N/A"
+
+  # Extract S3 bucket name from ARN
+  storage_bucket_name = split(":", module.agent.mcd_agent_storage_bucket_arn)[5]
+
+  # Common tags
+  common_tags = {
+    Service  = var.deployment_name
+    Provider = "monte-carlo"
+  }
+}
+
+# Monte Carlo Agent Module
+module "agent" {
+  source  = "monte-carlo-data/mcd-agent/aws"
+  version = "1.0.1"
+
+  cloud_account_id  = var.cloud_account_id
+  private_subnets   = var.existing_subnet_ids
+  image             = var.agent_image_uri
+  region            = var.region
+  remote_upgradable = var.remote_upgradable
+}
+
+# OpenTelemetry Collector Module
+module "opentelemetry_collector" {
+  source = "../aws_otel_collector"
```

Comment:
when `aws_otel_collector` directory is moved to its own repo and published as a Terraform module, we will change this `source` line to reference the published module.